Build, sign, distribute Android app via Firebase to testers

jamilbk opened the pull request:
Build, sign, and distribute Android app for testing.

Closes #2208

jamilbk committed:
Upload Android app bundle Take I

jamilbk committed:
Add commented-out if main

jamilbk committed:
Fix linter issues

jamilbk committed:
Use base64 properly

jamilbk committed:
use env var to specify testers

jamilbk committed:
Fix job specification

jamilbk committed:
Fix build/sign/upload order

jamilbk committed:
Write secrets just before they're used

jamilbk committed:
Print sha256 of secret files

jamilbk committed:
sha256sum secrets

jamilbk committed:
Inherit secrets DOH

jamilbk committed:
make sure PATH is in env

jamilbk committed:
Fix path

jamilbk committed:
Add --info for more info

jamilbk committed:
only on main

pratikvelani commented on kotlin/android/app/build.gradle.kts:
```diff
@@ -103,6 +116,14 @@ android {
+                serviceCredentialsFile = System.getenv("FIREBASE_CREDENTIALS_PATH")
+                artifactType = "AAB"
+                releaseNotes = "https://github.com/firezone/firezone/releases"
+                testers = System.getenv("FIREBASE_APP_TESTERS")
```
Can we use groups instead of individual tests here? 
https://firebase.google.com/docs/app-distribution/android/distribute-gradle#step_3_configure_your_distribution_properties

jamilbk replied:
Good call, updated!

jamilbk committed:
Use group instead of individual testers

jamilbk committed:
Fix rebase artifact

jamilbk committed:
Merge branch 'main' into ci/android-publish

jamilbk committed:
Merge branch 'main' into ci/android-publish

jamilbk committed:
Only upload app bundles with `distribute/apple` and `distribute/android` labels (#2358)

![Screenshot 2023-10-17 at 1 57 30
AM](https://github.com/firezone/firezone/assets/167144/894f86e7-ce76-412f-a09f-3e21e42246a5)
